lib/spack/spack/test/git_fetch.py

```diff
@@ -169,3 +172,48 @@ def test_needs_stage():
                        matches=_mock_transport_error):
         fetcher = GitFetchStrategy(git='file:///not-a-real-git-repo')
         fetcher.fetch()
+
+
+@pytest.mark.parametrize("get_full_repo", [True, False])
+def test_get_full_repo(get_full_repo, git_version, mock_git_repository,
+                       config, mutable_mock_packages):
+    """Ensure that we can clone a full repository."""
+
+    if git_version < ver('1.7.1'):
+        pytest.skip('Not testing get_full_repo for older git {0}'.
+                    format(git_version))
+
+    secure = True
+    type_of_test = 'tag-branch'
+
+    t = mock_git_repository.checks[type_of_test]
+
+    spec = Spec('git-test')
+    spec.concretize()
+    pkg = spack.repo.get(spec)
+    args = copy.copy(t.args)
+    args['get_full_repo'] = get_full_repo
+    pkg.versions[ver('git')] = args
+
+    with pkg.stage:
+        with spack.config.override('config:verify_ssl', secure):
+            pkg.do_stage()
+            with working_dir(pkg.stage.source_path):
+                branches\
+                    = mock_git_repository.git_exe('branch', '-a',
+                                                  output=str).splitlines()
+                nbranches = len(branches)
+                commits\
+                    = mock_git_repository.\
+                    git_exe('log', '--graph',
+                            '--pretty=format:%h -%d %s (%ci) <%an>',
+                            '--abbrev-commit',
+                            output=str).splitlines()
+                ncommits = len(commits)
+
+        if get_full_repo:
+            assert(nbranches == 5)
+            assert(ncommits == 2)
+        else:
+            assert(nbranches == 2)
```

Comment:
(question) why is this "2" rather than "1"?

Reply:
Local and remote branches: `[u'* tag-branch', u'  remotes/origin/tag-branch']`